tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -512,6 +503,23 @@ public async Task HandlePlaybackRequest(string recordingId, HttpRequest incoming
 
                 await WriteBodyBytes(bodyData, session.PlaybackResponseTime, outgoingResponse);
             }
+
+            Interlocked.Increment(ref Startup.RequestsPlayedBack);
+
+            // Only remove session once body has been written, to minimize probability client retries but test-proxy has already removed the session
+            var remove = true;
+
+            // If request contains "x-recording-remove: false", then request is not removed from session after playback.
+            // Used by perf tests to play back the same request multiple times.
+            if (incomingRequest.Headers.TryGetValue("x-recording-remove", out var removeHeader))
+            {
+                remove = bool.Parse(removeHeader);
+            }
+
+            if (remove)
+            {
+                session.Session.Lookup(entry, session.CustomMatcher ?? Matcher, sanitizers, remove: true);
```

Comment:
Might be able to improve perf, by adding something like `Session.Remove()`, which safely removes a session (under locks), but doesn't need to re-run the lookup and sanitizer code.

Reply:
👍 

I'm going to do that and publish this as a new proxy version to try reproing the storage issues on. I'm a bit concerned that the double sanitization won't be as stable as we expect (we sanitize the requestentry in `Lookup`), so I'm going to go after that Remove() separately.

Reply:
I agree adding a separate `Remove()` is the correct design, this was more a quick hack to show the direction I was thinking.